File: frontend/templates/react_clean_architecture/src/wire/interactor-factory.ts
import { DummyGateway } from "@template/dummy-gateway";
import {
  TemplateItem,
  TemplateItemId,
  TemplateItemTitle,
} from "@template/model";
import { ItemInteractor } from "@template/usecase";

export const buildItemInteractor = (): ItemInteractor =>
  new ItemInteractor(
    new DummyGateway([
      new TemplateItem(
        new TemplateItemId("first-item"),
        new TemplateItemTitle("Dummy item"),
      ),
    ]),
  );

File: mobile/templates/riverpod_clean_architecture/lib/wire/interactor_factory.dart
import 'package:dummy_gateway/dummy_gateway.dart';
import 'package:model/model.dart';
import 'package:usecase/usecase.dart';

ItemInteractor buildItemInteractor() => ItemInteractor(
  itemGateway: DummyGateway(
    initialItems: <TemplateItem>[
      TemplateItem(
        id: TemplateItemId('first-item'),
        title: TemplateItemTitle('Dummy item'),
      ),
    ],
  ),
);
